File: backend-fastapi/app/models/food.py
from datetime import datetime

from sqlalchemy import Float, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy.sql import func

from app.models import Base


class Food(Base):
    __tablename__ = "foods"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    name: Mapped[str | None] = mapped_column(String, nullable=True)
    category: Mapped[str | None] = mapped_column(String, nullable=True)
    weight: Mapped[float | None] = mapped_column(Float, nullable=True, comment="1회 기본 제공량(g)")
    calories: Mapped[float | None] = mapped_column(Float, nullable=True)
    protein: Mapped[float | None] = mapped_column(Float, nullable=True)
    carbs: Mapped[float | None] = mapped_column(Float, nullable=True)
    fat: Mapped[float | None] = mapped_column(Float, nullable=True)
    sodium: Mapped[float | None] = mapped_column(Float, nullable=True, comment="mg")
    saturated_fat: Mapped[float | None] = mapped_column(Float, nullable=True, comment="g")
    sugars: Mapped[float | None] = mapped_column("sugar", Float, nullable=True, comment="g")
    fiber: Mapped[float | None] = mapped_column("dietary_fiber", Float, nullable=True, comment="g")
    potassium: Mapped[float | None] = mapped_column(Float, nullable=True, comment="mg")
    purine_level: Mapped[str | None] = mapped_column(String, nullable=True)
    iodine: Mapped[float | None] = mapped_column(Float, nullable=True)
    selenium: Mapped[float | None] = mapped_column(Float, nullable=True)
    recommandable: Mapped[int] = mapped_column(Integer, nullable=False, default=0)

    profile: Mapped["FoodProfile | None"] = relationship(
        "FoodProfile", back_populates="food", uselist=False
    )


class FoodProfile(Base):
    __tablename__ = "food_profile"

    food_id: Mapped[int] = mapped_column(ForeignKey("foods.id"), primary_key=True)
    dish_role: Mapped[str | None] = mapped_column(String(20), nullable=True)
    food_group: Mapped[str | None] = mapped_column(String(20), nullable=True)
    gi: Mapped[int | None] = mapped_column(Integer, nullable=True)
    gl: Mapped[float | None] = mapped_column(Float, nullable=True)
    purine_mg: Mapped[float | None] = mapped_column(Float, nullable=True)
    updated_at: Mapped[datetime] = mapped_column(server_default=func.now(), onupdate=func.now())

    food: Mapped["Food"] = relationship("Food", back_populates="profile")

File: backend-spring/src/main/java/com/NurtiAgent/Onboard/food/entity/Food.java
package com.NurtiAgent.Onboard.food.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "foods")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Food {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", length = 100)
    private String name;

    @Column(name = "category", length = 100)
    private String category;

    @Column(name = "weight")
    private Double weight;

    @Column(name = "calories")
    private Integer calories;

    @Column(name = "protein")
    private Double protein;

    @Column(name = "carbs")
    private Double carbs;

    @Column(name = "fat")
    private Double fat;

    @Column(name = "sodium")
    private Double sodium;

    @Column(name = "saturated_fat")
    private Double saturatedFat;

    @Column(name = "sugar")
    private Double sugars;

    @Column(name = "dietary_fiber")
    private Double fiber;

    @Column(name = "cholesterol")
    private Double cholesterol;

    @Column(name = "trans_fat")
    private Double transFat;

    @Column(name = "potassium")
    private Double potassium;

    @Column(name = "purine_level", length = 10)
    private String purineLevel;

    @Column(name = "iodine")
    private Double iodine;

    @Column(name = "selenium")
    private Double selenium;
}
